spring-beans/src/main/java/org/springframework/beans/factory/xml/ResourceEntityResolver.java

```diff
@@ -110,27 +110,46 @@ public InputSource resolveEntity(@Nullable String publicId, @Nullable String sys
 				}
 			}
 			else if (systemId.endsWith(DTD_SUFFIX) || systemId.endsWith(XSD_SUFFIX)) {
-				// External dtd/xsd lookup via https even for canonical http declaration
-				String url = systemId;
-				if (url.startsWith("http:")) {
-					url = "https:" + url.substring(5);
-				}
-				try {
-					source = new InputSource(ResourceUtils.toURL(url).openStream());
-					source.setPublicId(publicId);
-					source.setSystemId(systemId);
-				}
-				catch (IOException ex) {
-					if (logger.isDebugEnabled()) {
-						logger.debug("Could not resolve XML entity [" + systemId + "] through URL [" + url + "]", ex);
-					}
-					// Fall back to the parser's default behavior.
-					source = null;
-				}
+				source = resolveEntityFallbackForSchemas(publicId, systemId);
 			}
 		}
 
 		return source;
 	}
 
+	/**
+	 * A fallback method for {@link #resolveEntity(String, String)} used when a "schema" entity (DTD or XSD) couldn't
+	 * be resolved as a local resource. The default behavior is to perform a remote resolution over https.
+	 * Override this method if this behavior is not desirable.
```

Comment:
```suggestion
	 * <p>Subclasses can override this method the change the default behavior.
	 * <ul>
	 * <li>Return {@code null} to fall back to the parser's
	 * {@linkplain org.xml.sax.EntityResolver#resolveEntity(String, String) default behavior}.</li>
	 * <li>Throw an exception to prevent remote resolution of the XSD or DTD.</li>
	 * </ul>
```